### ISSUE TITLE
Broken stored procedure body when using modern JavaScript syntax

### ISSUE THREAD

**Issue by bak-t** [PlagueHO/CosmosDB#137]
# Description
When using `Set-CosmosDbStoredProcedure`/`New-CosmosDbStoredProcedure` cmdlet to change/create stored procedure the body becomes broken.

For example code:
```javascript
...
if (someCondition)
  throw new Error(`Root entity (id: ${entity.id}) already exists`);
...
```
becomes
```javascript
...
if (someCondition)
  throw new Error(
oot entity (id: ${entity.id}) already exists`);
...
```

# Environment Details
* Powershell
  * Version: 5.1.15063.1155
  * Host: Console
* OS: Windows 10 Professional (10.0.15063)
* CosmosDB module version 2.1.2.514

# Additional observations
In `Set-CosmosDbStoredProcedure` stored procedure body is encoded using:
```powershell
$StoredProcedureBody = ((($StoredProcedureBody -replace '`n', '\n') -replace '`r', '\r') -replace '"', '\"')
```
The issue is in search string of `-replace` operation which should be enclosed in double quotes.
Also code to encode body can be replaced with 
```powershell
$StoredProcedureBody = ConvertTo-Json -InputObject $StoredProcedureBody
```
And even whole REST request body can be built like this:
```powershell
$requestBody = ConvertTo-Json -Depth 100 -InputObject @{
  id = $id
  body = $StoredProcedureBody
}
```
# P.S.
I hope to prepare pull request during next few days.

**Comment by PlagueHO** [comment]
Due to [this](https://github.com/PowerShell/PowerShell/issues/7693) difference between Windows PowerShell and PowerShell Core 6.0.x, this issue needs to be reopened. If this difference is addressed in PowerShell Core 6.1 then this issue can be closed again. Otherwise some additional steps will need to be taken to improve the conversion to work on PowerShell Core.

**Comment by PlagueHO** [comment]
Now that PowerShell Core 6.1 is long out we should be able to document and close this issue by defining a minimum recommended PS version and ensuring the tests work correctly with it.
